executor/executor.go

```diff
@@ -571,6 +594,51 @@ func (e *SelectionExec) Next(goCtx goctx.Context) (Row, error) {
 	}
 }
 
+// NextChunk implements the Executor NextChunk interface.
+func (e *SelectionExec) NextChunk(chk *chunk.Chunk) error {
+	chk.Reset()
+	for {
+		for ; e.inputRow != e.childrenResults[0].End(); e.inputRow = e.inputRow.Next() {
+			if e.batched {
```

Comment:
I think you can split batch  and non-batch logic into two different function.
Keep the function small and compact make the code more readable.